File: Examples/Chart/WebForm/VB.NET/Examples/Panels/Annotations/NPieDataPointAnchorUC.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports Nevron.Examples.Framework.WebForm
Imports Nevron.GraphicsCore
Imports Nevron.Chart
Imports Nevron.Chart.WebForm

Namespace Nevron.Examples.Chart.WebForm
	Public Partial Class NPieDataPointAnchorUC
		Inherits NExampleUC
		Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
			nChartControl1.BackgroundStyle.FrameStyle.Visible = False

			nChartControl1.Panels.Clear()

			' Create title label
			Dim title As NLabel = New NLabel("Pie Data Point Anchor")
			title.TextStyle.FontStyle = New NFontStyle("Times New Roman", 18, FontStyle.Italic)
			title.TextStyle.FontStyle = New NFontStyle("Times New Roman", 14, FontStyle.Italic)
			title.TextStyle.ShadowStyle.Type = ShadowType.LinearBlur
			nChartControl1.Panels.Add(title)

			' Create a pie chart
			Dim chart As NPieChart = New NPieChart()
			nChartControl1.Panels.Add(chart)
			chart.Enable3D = False

			' Create a pie series with 6 data points
			Dim pieSeries As NPieSeries = New NPieSeries()
			chart.Series.Add(pieSeries)
			pieSeries.DataLabelStyle.Visible = True
			pieSeries.LabelMode = PieLabelMode.SpiderNoOverlap
			pieSeries.Values.FillRandomRange(New Random(), 6, 1, 5)

			' Create a rounded rect callout
			Dim callout As NRoundedRectangularCallout = New NRoundedRectangularCallout()
			callout.ArrowLength = New NLength(20, NGraphicsUnit.Point)
			callout.FillStyle = New NGradientFillStyle(GradientStyle.Horizontal, GradientVariant.Variant1, Color.FromArgb(125, Color.White), Color.FromArgb(125, Color.LightGreen))
			callout.UseAutomaticSize = True
			callout.Orientation = 80
			callout.ContentAlignment = ContentAlignment.TopLeft
			callout.Text = "Annotation"
			callout.TextStyle.FontStyle.EmSize = New NLength(8)

			' Anchor the callout to pie data point #0
			Dim anchor As NPieDataPointAnchor = New NPieDataPointAnchor(pieSeries, 0, 0.8f, StringAlignment.Near)
			callout.Anchor = anchor

			' add the annotation panel
			nChartControl1.Panels.Add(callout)

			' apply style sheet
			Dim styleSheet As NStyleSheet = NStyleSheet.CreatePredefinedStyleSheet(PredefinedStyleSheet.BrightMultiColor)
			styleSheet.Apply(nChartControl1.Document)
		End Sub
	End Class
End Namespace

File: Examples/Chart/WebForm/CSharp/Examples/Panels/Annotations/NPieDataPointAnchorUC.ascx.cs
using System;
using System.Drawing;
using System.Web.UI;
using System.Web.UI.WebControls;
using Nevron.Examples.Framework.WebForm;
using Nevron.GraphicsCore;
using Nevron.Chart;
using Nevron.Chart.WebForm;

namespace Nevron.Examples.Chart.WebForm
{
	public partial class NPieDataPointAnchorUC : NExampleUC
	{
		protected void Page_Load(object sender, EventArgs e)
		{
            nChartControl1.BackgroundStyle.FrameStyle.Visible = false;

            nChartControl1.Panels.Clear();

            // Create title label
			NLabel title = new NLabel("Pie Data Point Anchor");
			title.TextStyle.FontStyle = new NFontStyle("Times New Roman", 18, FontStyle.Italic);
            title.TextStyle.FontStyle = new NFontStyle("Times New Roman", 14, FontStyle.Italic);
            title.TextStyle.ShadowStyle.Type = ShadowType.LinearBlur;
            nChartControl1.Panels.Add(title);

			// Create a pie chart
			NPieChart chart = new NPieChart();
            nChartControl1.Panels.Add(chart);
			chart.Enable3D = false;

			// Create a pie series with 6 data points
			NPieSeries pieSeries = new NPieSeries();
			chart.Series.Add(pieSeries);
			pieSeries.DataLabelStyle.Visible = true;
			pieSeries.LabelMode = PieLabelMode.SpiderNoOverlap;
            pieSeries.Values.FillRandomRange(new Random(), 6, 1, 5);

			// Create a rounded rect callout
			NRoundedRectangularCallout callout = new NRoundedRectangularCallout();
			callout.ArrowLength = new NLength(20, NGraphicsUnit.Point);
			callout.FillStyle = new NGradientFillStyle(GradientStyle.Horizontal, GradientVariant.Variant1, Color.FromArgb(125, Color.White), Color.FromArgb(125, Color.LightGreen));
			callout.UseAutomaticSize = true;
			callout.Orientation = 80;
			callout.ContentAlignment = ContentAlignment.TopLeft;
			callout.Text = "Annotation";
			callout.TextStyle.FontStyle.EmSize = new NLength(8);

			// Anchor the callout to pie data point #0
			NPieDataPointAnchor anchor = new NPieDataPointAnchor(pieSeries, 0, 0.8f, StringAlignment.Near);
			callout.Anchor = anchor;
            
			// add the annotation panel
			nChartControl1.Panels.Add(callout);

			// apply style sheet
			NStyleSheet styleSheet = NStyleSheet.CreatePredefinedStyleSheet(PredefinedStyleSheet.BrightMultiColor);
            styleSheet.Apply(nChartControl1.Document);
		}
	}
}
